### PR TITLE
[4.0] Use identical operators for plugins

### PR DESCRIPTION
**Pull request by C-Lodder**
### Summary of Changes

This PR uses identical operators where possible for the `plugins` directory. Just a minor performance improvement.

### Testing Instructions

Code review

### CODE REVIEW

**Review by mbabker — APPROVED** [review]

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 827839f175bcaa6a4d9a53413d4a400f42495ca2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19699">issues.joomla.org/tracker/joomla-cms/19699</a>.</sub>

**Comment by anuragteapot** [comment]
I have tested this item :white_check_mark: successfully on 827839f175bcaa6a4d9a53413d4a400f42495ca2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19699">issues.joomla.org/tracker/joomla-cms/19699</a>.</sub>

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

File: plugins/fields/calendar/tmpl/calendar.php [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 $value = $field->value;
 
-if ($value == '')
+if ($value === '')
```
Comment (laoneo) [inline plugins/fields/calendar/tmpl/calendar.php]: Are you sure we do not break here something as 0 and false are allowed now? Same for the others where you do === ''.

**Comment by brianteeman** [comment]
@C-Lodder can you please respond to the question from @laoneo and then we can move forward with this.

File: plugins/fields/calendar/tmpl/calendar.php (hunk 1 above)
Comment (C-Lodder) [reply plugins/fields/calendar/tmpl/calendar.php]: Surely `''` isnt the same as 0 or false?

Comment (laoneo) [reply plugins/fields/calendar/tmpl/calendar.php]: If you do strict comparison, then no.

**Comment by laoneo** [comment]
Honestly I would revert all the `=== ''` checks as they need to be checked on a case by case basis, because it can have unwanted side effects.

**Comment by ggppdk** [comment]
About strict comparison in template files (of custom fields)

> Are you sure we do not break here something as 0 and false are allowed now? Same for the others where you do === ''.

Actually this change will allow an integer value of 0 (zero)
because current comparison is unable to display zero (i mean the integer type of zero)

but then you would ask, why currently display of zero is possible ? (i mean currently it works)

It is because currently fields like
- text, textarea, integer fields
use default handling for values (they do not implement some custom handling before passing to the value to the template),
which passes to the template file (/tmpl/*.php) a string value

That is this is passed
- `string(1) "0"`

thus this if (`string(1) "0"` == '')   evaluates to false, thus display of zero is possible

The strict comparison will effect the case of passing a value NULL or FALSE
since it will allow them to be "displayed"
but then current template code for such value will show nothing anyway
aka the current display of fields will not change

**Comment by C-Lodder** [comment]
@laoneo @brianteeman all the `== ''` have been reverted as request. Please squash the commits

**Comment by laoneo** [comment]
Thanks

**Pull request merged**